rpc/tests/utils/tiny_node.rs

```diff
@@ -0,0 +1,100 @@
+//! Tiny p2p node without any defined behaviour celestia can connect to so that we can test RPC p2p
+//! calls
+
+use celestia_rpc::p2p;
+use futures::StreamExt;
+use libp2p::{
+    core::upgrade::Version,
+    identify,
+    identity::{self, Keypair},
+    noise,
+    swarm::{keep_alive, NetworkBehaviour, SwarmBuilder, SwarmEvent},
+    tcp, yamux, Multiaddr, PeerId, Transport,
+};
+use tokio::{
+    task::JoinHandle,
+    time::{sleep, Duration},
+};
+
+// how long to wait during startup for node to start listening on interfaces, before we return a
+// list of addresses
+const NODE_ADDRESS_ACQUIRE_DELAY_TIME: Duration = Duration::from_millis(100);
+
+/// Our network behaviour.
+#[derive(NetworkBehaviour)]
+struct Behaviour {
+    identify: identify::Behaviour,
+    keep_alive: keep_alive::Behaviour,
+}
+
+impl Behaviour {
+    fn new(local_key: Keypair) -> Self {
+        let identify =
+            identify::Behaviour::new(identify::Config::new("".to_owned(), local_key.public()));
+
+        Self {
+            identify,
+            keep_alive: keep_alive::Behaviour,
+        }
+    }
+}
+
+pub async fn start_tiny_node() -> anyhow::Result<(p2p::AddrInfo, JoinHandle<()>)> {
+    async fn get_addresses<T: NetworkBehaviour>(swarm: &mut libp2p::Swarm<T>) -> Vec<Multiaddr> {
```

Comment:
do not define a function within a function

Reply:
I wanted to keep them together since this function is mostly for code separation, but I don't have strong feelings about moving it. 
Is it particular objection to using nested function here or in general?

Reply:
In general. It makes reading the code harder and the function look larger than they are.

Reply:
> Is it particular objection to using nested function here or in general?

yeah, it's rather rare to do so